src/unused_finder/unused_finder.rs

```diff
@@ -0,0 +1,323 @@
+            .filter_map(|file| {
+                if self.entry_packages.contains(&file.package_name) {
+                    return Some(file.source_file_path.clone());
+                }
+                None
+            })
+            .collect();
+        let mut frontier = entry_files;
+        for i in 0..10_000_000 {
+            frontier = self.graph.bfs_step(frontier);
+
+            if frontier.is_empty() {
+                break;
+            }
+            if i == 10_000_000 {
+                return Err(napi::Error::new(
+                    napi::Status::GenericFailure,
+                    "exceeded max iterations".to_string(),
+                ));
+            }
+        }
+
+        let allow_list: Vec<glob::Pattern> = read_allow_list();
+
+        let reported_unused_files = BTreeMap::from_iter(
+            self.graph
+                .files
+                .iter()
+                .filter(|f| !f.1.is_used && !allow_list.iter().any(|p| p.matches(f.0))),
+        );
+        let unused_files_items: HashMap<String, Vec<ExportedItemReport>> = self
+            .graph
+            .files
+            .iter()
+            .filter_map(|(file_path, info)| {
+                if info.is_used {
+                    match file_path_exported_items_map.remove(file_path) {
+                        Some(mut exported_items) => {
+                            let unused_exports = &info.unused_exports;
+                            if unused_exports.is_empty() {
+                                return None;
+                            }
+                            let unused_exports = exported_items
+                                .drain(0..)
+                                .filter(|exported| {
+                                    unused_exports
+                                        .iter()
+                                        .any(|unused| unused.to_string() == exported.id.to_string())
+                                })
+                                .collect();
+                            return Some((file_path.to_string(), unused_exports));
+                        }
+                        None => return None,
+                    }
+                }
+                None
+            })
+            .collect();
+
+        let mut ok = UnusedFinderReport {
+            unused_files: reported_unused_files
+                .iter()
+                .map(|(p, _)| p.to_string())
+                .collect(),
+            unused_files_items,
+        };
+        let files: HashSet<String> = HashSet::from_iter(files_to_check);
+        ok.unused_files_items.retain(|key, _| files.contains(key));
+        return Ok(ok);
+    }
+
+    fn refresh_files_to_check(&mut self, files_to_check: &Vec<String>) {
+        files_to_check.iter().for_each(|f| {
+            match self
+                .flattened_walk_file_data
+                .par_iter_mut()
+                .find_first(|walk_file_info| &walk_file_info.source_file_path == f)
```

Comment:
This does not update the representation within the grpah, so the 2 sources of "file state" are now out of sync